src/domains/ecommerce-common/03_storage.tf

```diff
@@ -169,4 +169,421 @@ resource "azurerm_storage_queue" "transactions_dead_letter_queue" {
 resource "azurerm_storage_queue" "notifications_service_errors_queue" {
   name                 = "${local.project}-notifications-service-errors-queue"
   storage_account_name = module.ecommerce_storage_deadletter.name
+}
+
+# Ecommerce transient queue alert
+resource "azurerm_monitor_diagnostic_setting" "ecommerce_transient_queue_diagnostics" {
+  count                      = var.env_short == "p" ? 1 : 0
+  name                       = "${module.ecommerce_storage_transient.name}-diagnostics"
+  target_resource_id         = "${module.ecommerce_storage_transient.id}/queueServices/default/"
+  log_analytics_workspace_id = data.azurerm_log_analytics_workspace.log_analytics.id
+
+  enabled_log {
+    category = "StorageWrite"
+
+    retention_policy {
+      enabled = true
+      days    = 7
+    }
+  }
+
+  enabled_log {
+    category = "StorageDelete"
+
+    retention_policy {
+      enabled = true
+      days    = 7
+    }
+  }
+}
+
+# Queue size: Ecommerce - ecommerce transactions expiration queue size
+resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_transactions_expiration_queue_size" {
+  count               = var.env_short == "p" ? 1 : 0
+  name                = "${local.project}-ecommerce-transactions-expiration-queue-size-alert"
+  resource_group_name = azurerm_resource_group.storage_ecommerce_rg.name
+  location            = var.location
+
+  action {
+    action_group           = [data.azurerm_monitor_action_group.email.id, data.azurerm_monitor_action_group.slack.id]
+    email_subject          = "Email Header"
+    custom_webhook_payload = "{}"
+  }
+  data_source_id = module.ecommerce_storage_transient.id
+  description    = "Ecommerce expirations queue size > 10"
+  enabled        = true
+  query = format(<<-QUERY
+    let OpCountForQueue = (operation: string, queueKey: string) {
+      StorageQueueLogs
+      | where OperationName == operation and ObjectKey startswith queueKey
+      | summarize count()
+    };
+    let MessageRateForQueue = (queueKey: string) {
+      OpCountForQueue("PutMessage", queueKey)
+      | join OpCountForQueue("DeleteMessage", queueKey) on count_
+      | project name = queueKey, Count = count_ - count_1
+    };
+    MessageRateForQueue("%s")
+    QUERY
+    , format("%s/pagopa-%s-weu-%s", module.ecommerce_storage_transient.name, var.env_short, "ecommerce-transactions-expiration-queue")
```

Comment:
You can avoid repetition by using a [for_each](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each) argument inside the resource